test/unit/alphabet/nucleotide/sam_dna16_test.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <seqan3/alphabet/nucleotide/sam_dna16.hpp>
+
+#include <seqan3/io/stream/parse_condition.hpp>
+
+#include "../alphabet_test_template.hpp"
+#include "../alphabet_constexpr_test_template.hpp"
+
+using namespace seqan3;
+
+// ------------------------------------------------------------------
+// sam_eq alphabet
+// ------------------------------------------------------------------
+
+INSTANTIATE_TYPED_TEST_CASE_P(sam_eq, alphabet, sam_eq);
+INSTANTIATE_TYPED_TEST_CASE_P(sam_eq, alphabet_constexpr, sam_eq);
+
+TEST(sam_eq, to_char_assign_char)
+{
+    EXPECT_EQ(to_char(sam_eq{}.assign_char('=')), '=');
+    EXPECT_EQ(to_char(sam_eq{}.assign_char('A')), '=');
+}
+
+// ------------------------------------------------------------------
+// sam_dna16 alphabet
+// ------------------------------------------------------------------
+
+INSTANTIATE_TYPED_TEST_CASE_P(sam_dna16, alphabet, sam_dna16);
+INSTANTIATE_TYPED_TEST_CASE_P(sam_dna16, alphabet_constexpr, sam_dna16);
+
+// nucleotide test: (because the complement is not bijective for sam_dna16 we need to test in manually)
+TEST(sam_dna16, nucleotide)
+{
+    EXPECT_TRUE(NucleotideAlphabet<sam_dna16>);
+    EXPECT_TRUE(NucleotideAlphabet<sam_dna16 &>);
+
+    EXPECT_EQ(complement('A'_sam_dna16), 'T'_sam_dna16);
+    EXPECT_EQ(complement('C'_sam_dna16), 'G'_sam_dna16);
+    EXPECT_EQ(complement('G'_sam_dna16), 'C'_sam_dna16);
+    EXPECT_EQ(complement('T'_sam_dna16), 'A'_sam_dna16);
+
+    EXPECT_EQ(complement('='_sam_dna16), 'N'_sam_dna16);
+}
+
+TEST(sam_dna16, to_char_assign_char)
+{
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('A')), 'A');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('C')), 'C');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('G')), 'G');
+
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('U')), 'T');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('T')), 'T');
+
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('R')), 'R');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('Y')), 'Y');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('S')), 'S');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('W')), 'W');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('K')), 'K');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('M')), 'M');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('B')), 'B');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('D')), 'D');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('H')), 'H');
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('V')), 'V');
+
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('=')), '=');
+
+    EXPECT_EQ(to_char(sam_dna16{}.assign_char('N')), 'N');
```

Comment:
for most of them you can use a loop
`for (char chr : std::string{"ACGTRY..."})`